### DISCUSSION TITLE
How to inject static routing information to Cisco

### DISCUSSION THREAD

**Discussion by areksm** [strongswan/strongswan#2020] [General]
Hi,
My Strongswan IPsec device (as a spoked) is connected to a Cisco router (HUB) with FlexVPN (IKEv2).
I am using a Strongswan's route based policy with VTI interfaces.
First I set the traffic selectors to 0.0.0.0/0.
And it works well.

However, our client added additional requirements for running BGP between peers in an Ipsec tunnel.
But BGP needs the peer's IP address.
I can of course set static routing on my end, but how to do it dynamically on Cisco FlexVPN (such a requirement).
I tried extending the local traffic selectors with my peer IP address, but during the ikev2 Strongswan session only the traffic selector 0.0.0.0/0 was sent.
 
Is there any way to send static routing in ikev2 exchange from Strongswan to Cisco?

I thought:
a) I found in Cisco command "route set interface:
"Create an IKEv2 authorization policy, the command set interface will send the tunnel IP address as a static IP address to the peer device."

Is there anything similar on Strongwan?
I couldn't find it in the documentation and examples

b) I also thought about a configuration payload with the INTERNAL_IP4_SUBNET attribute:
https://www.cisco.com/c/en/us/td/docs/ios-xml/ios/sec_conn_ike2vpn/configuration/xe-3s/sec-flex-vpn-xe-3s-book/sec-cfg- flex-serv.html.

But here it doesn't seem right because our device is the IKE Initiator and I'm not sure if it can send CFG_SET to Cisco.

Additionally, on the StrongSwan website (https://docs.strongswan.org/docs/5.9/plugins/attr.html) it says:

“Configured attributes are assigned to peers via CP configuration payloads (IKEv2) or via mode configuration (IKEv1).
Attributes are only assigned to partners when they request a virtual IP address." 
But virtual IP address is not in my case.

c) Cisco has something called Reverse Route Injection, but I don't know how to integrate with Strongswan.

To summarize:
How to send information of static routing from Strongswan to Cisco when selectors are set 0.0.0.0/0.

Arek
